### ISSUE TITLE
Remove background color from hinted elements to keep original styling

### ISSUE THREAD

**Issue by Jeusto** [tridactyl/tridactyl#4106]
Tridactyl version: 1.22.0
Firefox version: Mozilla Firefox 97.0
Operating system: linux

Hi, I wanted to create my own theme for Tridactyl and I've stumbled upon a problem. In hint mode, I couldn't leave elements like buttons with their original background color. This is a bit annoying for some websites with a lot buttons. I've tried styling the class "TridactylHintElement" or the root variable "--tridactyl-hint-bg" but with no success. I have to either choose a custom background color or have no background color at all, I would like to keep the original background color for all hinted elements.

Here's how it looks and how I would like to make it look like.
![example1](https://user-images.githubusercontent.com/65134804/154115706-ed23d7cf-4a77-4f12-bead-bc8090bfaa9c.png)

![example2](https://user-images.githubusercontent.com/65134804/154115716-71eaa05a-64fe-4888-a1ae-ae93f4a56c22.png)

Also another unrelated thing: there's always a slight delay when opening new tabs where the default styling appears and then it changes to what it should be. Would it be possible to disable the css injection that the extension does, and leave it to the user to add the styling in the "userChrome.css" file?

**Comment by bovine3dom** [comment]
I can't think of an obvious way of doing what you want, but maybe someone else will.


> there's always a slight delay when opening new tabs where the default styling appears and then it changes to what it should be. Would it be possible to disable the css injection that the extension does, and leave it to the user to add the styling in the "userChrome.css" file?


See https://github.com/tridactyl/tridactyl/issues/2510#issuecomment-763198138 - I think that solves your problem :)

**Comment by davidscotson** [comment]
I don't think this is possible to do from a theme with the way the current core themeing code works.

Because we use CSS vars, it's impossible to set a broken or null value, it'll just default to setting the background to transparent.

If it was the color rather than the background-color we wanted, then using 'currentColor' would work, but we only are able to override, and I don't think there's any way to get the current background color (see https://github.com/w3c/csswg-drafts/issues/5292)

So probably the core theme code needs tweaked in some way to make this possible.

I was looking at using ::after to create overlays rather than directly affecting the page content when styling hints, but styling injected code on a random webpage is pretty wild, seems there's lots of gotchas to look out for. But I'll report back if it seems to be working out, as it gives some extra flexability.

**Comment by Jeusto** [comment]
Yeah I've quickly looked into it and "::after" works pretty well if the "hint-bg" is a color with some transparency since it goes over the element. Thanks for looking into it.

**Comment by bovine3dom** [comment]
We can probably do it fairly easily with a separate stylesheet and `browser.tabs.insertCSS` that just injects this line https://github.com/tridactyl/tridactyl/blob/78d0fa91df0d76232f71fde39cadf681dff44d39/src/static/css/hint.css#L9
 if some config flag is set to true.

**Comment by rddunphy** [comment]
I have a related problem with my theme. I'm not trying to disable the background colour altogether, as I like being able to see which hint is selected - instead, I've set the background colours to values with low opacity, so that the elements stand out less. This looks all right in most cases:

![hint_bg1](https://user-images.githubusercontent.com/10012625/210620011-0d2f617b-8a88-47b8-a521-781e90330a7c.png)

But the trade-off is that it's extra horrible in hint modes that select all divs if the site uses a dark background:

![hint_bg2](https://user-images.githubusercontent.com/10012625/210620025-2d28ee41-e195-490a-a618-8893c70b8a61.png)

I think overlay divs would be a nicer solution than a binary setting.

**Comment by bovine3dom** [comment]
I'm very far from an expert on styling and CSS but I don't understand how overlaying the elements with divs and styling that would make any difference - wouldn't we just put a div over the entire page and style that, with exactly the same result visually?

**Comment by bovine3dom** [comment]
Thanks, I get it now. That looks like a good idea, let's do that :)

**Comment by Chic-Tweetz** [comment]
I've added a config object, `hintstyles` with keys `bg`, `fg`, and `outline` which can be set to `"all"`, "active" or "none" to decide which rules to add.
So you could for example have outlines on all hints, but only a background for the active hint, using `set hintstyles.bg active` and `set hintstyles.outline all`.
I use `browserBg.tabs.insertCSS` to insert a string built according to the config, basically copying how theme styles are inserted in styling.ts.

**Comment by bovine3dom** [comment]
Fixed in #5097: `:set hintstyles.bg active` in 1.24.2+ (unreleased at time of writing)

**Comment by YousufSSyed** [comment]
Yep, for anyone reading this, you can disable all the hinted element styling like so

```vim
" Remove all styling from hints
set hintstyles.bg none
set hintstyles.fg none
set hintstyles.outline none
```
